Microsoft.VCRTForwarders.140.targets

```diff
@@ -4,34 +4,91 @@
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x64\native\release\*.dll" />
-  </ItemGroup>
-   <ItemGroup Condition="'$(Configuration)' == 'Release' And '$(Platform)' == 'x64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-x64\native\release\*.dll" />
-  </ItemGroup>
+        var properties = new Dictionary<string, string>
+        {
+          { "Configuration", Configuration },
+          { "Platform", Platform }
+        };
+        
+        // Per MSDN, _DEBUG define can be checked to determine if debug CRT is in use.
+        var collection = new ProjectCollection(properties);
+        var project = collection.LoadProject(ProjectFile);
+        ProjectMetadata def = project.AllEvaluatedItemDefinitionMetadata
+                                     .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "PreprocessorDefinitions");
+        Boolean useDebug = def != null && (";" + def.EvaluatedValue + ";").Contains(";_DEBUG;");
 
-  <!-- arm64 -->
-  <ItemGroup Condition="'$(Configuration)' == 'Debug' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\debug\*.dll" />
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
-   <ItemGroup Condition="'$(Configuration)' == 'Release' And '$(Platform)' == 'ARM64'">
-    <ReferenceCopyLocalPaths Include="$(MSBuildThisFileDirectory)..\..\runtimes\win10-arm64\native\release\*.dll" />
-  </ItemGroup>
+        // There seem to be cases where even if _DEBUG is not found, debug CRT is used based on RuntimeLibrary.
+        if(!useDebug)
+        {
+          ProjectMetadata runtimeLibrary = project.AllEvaluatedItemDefinitionMetadata
+                                                  .LastOrDefault(p => p.ItemType == "ClCompile" && p.Name == "RuntimeLibrary");
+          useDebug = runtimeLibrary != null && (String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebugDLL", true) == 0 ||
+                                                String.Compare(runtimeLibrary.EvaluatedValue, "MultiThreadedDebug", true) == 0);
+        }
+
+        TaskOutput = useDebug;
+        
+        ]]>
+      </Code>
+    </Task>
+  </UsingTask>
+  
+  <Target Name="ResolveVCRTForwarderReferences">
+    <UseDebugCRT ProjectFile="%(ProjectReferenceWithConfiguration.Identity)"
+                 Configuration="%(ProjectReferenceWithConfiguration.Configuration)"
+                 Platform="%(ProjectReferenceWithConfiguration.Platform)"
+                 Condition="'@(ProjectReferenceWithConfiguration)' != '' And '$(VCRTForwarders-IncludeDebugCRT)' == ''">
+      <Output ItemName="TaskOutput" TaskParameter="TaskOutput"/>
+    </UseDebugCRT>
+    <PropertyGroup>
+      <VCRTForwarders-IncludeDebugCRT Condition="'%(TaskOutput.Identity)' == 'true'">true</VCRTForwarders-IncludeDebugCRT>
```

Comment:
you can set VCRTForwarders-IncludeDebugCRT directly from the output of your task

Reply:
The task can be ran multiple times based on # of references and I don't want a false result overriding a previous true result.  I can make the condition check for the property first, but I noticed before that sometimes the tasks seem to run in parallel causing them to be overridden.  That was why I had it outside as a separate property rather than using output to set it.